diff --git a/cola/Makefile.am b/cola/Makefile.am
index 180f2ae9..c902ad5c 100644
--- a/cola/Makefile.am
+++ b/cola/Makefile.am
@@ -1,4 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4
 AUTOMAKE_OPTIONS = gnu
-SUBDIRS = libavoid libvpsc libcola libtopology libdialect
-
+SUBDIRS = libavoid libvpsc libcola libtopology
+if DIALECT
+  SUBDIRS += libdialect
+endif
diff --git a/cola/configure.ac b/cola/configure.ac
index a79e1a59..d2d15678 100644
--- a/cola/configure.ac
+++ b/cola/configure.ac
@@ -8,7 +8,6 @@ AC_ENABLE_STATIC
 AC_ENABLE_SHARED
 AC_PROG_CXX
 AC_PROG_CC
-AC_PROG_LIBTOOL
 LT_INIT
 AC_PROG_INSTALL
 #AC_DEFINE(TRACE_LOGGING)
@@ -61,5 +60,20 @@ AH_BOTTOM([
   
 ])
 
-AC_CONFIG_FILES([Makefile libcola/Makefile libcola/tests/Makefile libvpsc/Makefile libvpsc/tests/Makefile libtopology/Makefile libtopology/tests/Makefile libavoid/Makefile libavoid/tests/Makefile libdialect/Makefile libdialect/tests/Makefile libcola/libcola.pc libavoid/libavoid.pc libvpsc/libvpsc.pc libtopology/libtopology.pc libdialect/libdialect.pc])
+AC_CONFIG_FILES([Makefile libcola/Makefile libcola/tests/Makefile libvpsc/Makefile libvpsc/tests/Makefile libtopology/Makefile libtopology/tests/Makefile libavoid/Makefile libavoid/tests/Makefile libcola/libcola.pc libavoid/libavoid.pc libvpsc/libvpsc.pc libtopology/libtopology.pc])
+
+AC_ARG_ENABLE([dialect],
+  AS_HELP_STRING(
+    [--disable-dialect],
+    [Turn off libdialect @<:@default=no@:>@]
+  ),
+  AS_CASE(${enableval},
+    [yes], [dialect=true],
+    [no], [dialect=false],
+    AC_MSG_ERROR([bad value ${enableval} for --disable-dialect])
+  ),
+  [dialect=false])
+AM_CONDITIONAL([DIALECT], [test x$dialect = xtrue])
+AM_COND_IF([DIALECT], [AC_CONFIG_FILES([libdialect/Makefile libdialect/tests/Makefile libdialect/libdialect.pc])])
+
 AC_OUTPUT
diff --git a/cola/libavoid/vertices.h b/cola/libavoid/vertices.h
index aac43bb4..7bc6240e 100644
--- a/cola/libavoid/vertices.h
+++ b/cola/libavoid/vertices.h
@@ -107,8 +107,8 @@ class VertID
 // orthogonal visibility graph since the vertices in the orthogonal graph 
 // mostly do not correspond to shape corners or connector endpoints.
 //
-static const VertID dummyOrthogID(0, 0);
-static const VertID dummyOrthogShapeID(0, 0, VertID::PROP_OrthShapeEdge);
+static const VertID dummyOrthogID{0, 0};
+static const VertID dummyOrthogShapeID{0, 0, VertID::PROP_OrthShapeEdge};
 
 class ANode;
 
diff --git a/cola/libcola/tests/Makefile.am b/cola/libcola/tests/Makefile.am
index a73ff824..5f9b1edb 100644
--- a/cola/libcola/tests/Makefile.am
+++ b/cola/libcola/tests/Makefile.am
@@ -38,7 +38,6 @@ unsatisfiable_SOURCES = unsatisfiable.cpp
 
 invalid_SOURCES = invalid.cpp 
 
-makefeasible_LDADD = $(LDADD) $(top_srcdir)/libavoid/libavoid.la
 makefeasible_SOURCES = makefeasible02.cpp 
 
 shortest_paths_SOURCES = shortest_paths.cpp
@@ -55,7 +54,7 @@ shortest_paths_SOURCES = shortest_paths.cpp
 #convex_hull_SOURCES = convex_hull.cpp
 #convex_hull_LDADD = $(LDADD) -lcairo -lcairomm-1.0
 #aarontest_SOURCES = aarontest.cpp
-#aarontest_LDADD = $(top_srcdir)/libcola/libcola.la $(top_builddir)/libvpsc/libvpsc.la -lcairo -lcairomm-1.0
+#aarontest_LDADD = $(LDADD) -lcairo -lcairomm-1.0
 #qo1_SOURCES = qo1.c
 #qo1_LDADD = -L$(mosek_home)/bin -lmosek -lguide -limf
 #small_graph_SOURCES = small_graph.cpp 
diff --git a/cola/libdialect/tests/Makefile.am b/cola/libdialect/tests/Makefile.am
index d2e5a42d..b19d25d6 100644
--- a/cola/libdialect/tests/Makefile.am
+++ b/cola/libdialect/tests/Makefile.am
@@ -104,11 +104,11 @@ treeplacement03_SOURCES = treeplacement03.cpp
 trees_SOURCES = trees.cpp
 trees2_SOURCES = trees2.cpp
 vpsc01_SOURCES = vpsc01.cpp
-holasbgn01_SOURCES = holasbgn01.cpp
-holasbgn02_SOURCES = holasbgn02.cpp
-holametro01_SOURCES = holametro01.cpp
-holasbgn03_SOURCES = holasbgn03.cpp
-holasbgn04_SOURCES = holasbgn04.cpp
-holaRand_SOURCES = holaRand.cpp
+#holasbgn01_SOURCES = holasbgn01.cpp
+#holasbgn02_SOURCES = holasbgn02.cpp
+#holametro01_SOURCES = holametro01.cpp
+#holasbgn03_SOURCES = holasbgn03.cpp
+#holasbgn04_SOURCES = holasbgn04.cpp
+#holaRand_SOURCES = holaRand.cpp
 
 TESTS = $(check_PROGRAMS)
