diff --git a/Examples/javascript/package.json b/Examples/javascript/package.json
new file mode 100644
index 000000000..0967ef424
--- /dev/null
+++ b/Examples/javascript/package.json
@@ -0,0 +1 @@
+{}
diff --git a/Examples/test-suite/common.mk b/Examples/test-suite/common.mk
index fee130277..46ad92d1a 100644
--- a/Examples/test-suite/common.mk
+++ b/Examples/test-suite/common.mk
@@ -86,8 +86,14 @@ CPP_TEST_BROKEN += \
 	director_nested_class \
 	exception_partial_info \
 	extend_variable \
+	li_boost_shared_ptr \
+	li_boost_shared_ptr_attribute \
+	li_boost_shared_ptr_bits \
+	li_boost_shared_ptr_director \
 	li_boost_shared_ptr_template \
+	multiple_inheritance_shared_ptr \
 	nested_private \
+	preproc_include \
 	template_default_pointer \
 	$(CPP11_TEST_BROKEN) \
 	$(CPP14_TEST_BROKEN) \
@@ -291,11 +297,6 @@ CPP_TEST_CASES += \
 	langobj \
 	li_attribute \
 	li_attribute_template \
-	li_boost_shared_ptr \
-	li_boost_shared_ptr_attribute \
-	li_boost_shared_ptr_bits \
-	li_boost_shared_ptr_director \
-	li_boost_shared_ptr_template \
 	li_carrays_cpp \
 	li_cdata_cpp \
 	li_cpointer_cpp \
@@ -319,7 +320,6 @@ CPP_TEST_CASES += \
 	multiple_inheritance_interfaces \
 	multiple_inheritance_nspace \
 	multiple_inheritance_overload \
-	multiple_inheritance_shared_ptr \
 	name_cxx \
 	name_warnings \
 	namespace_chase \
@@ -826,7 +826,6 @@ C_TEST_CASES += \
 	preproc_defined \
 	preproc_expr \
 	preproc_gcc_output \
-	preproc_include \
 	preproc_line_file \
 	preproc_predefined \
 	preproc_predefined_stdc \
diff --git a/Examples/test-suite/director_smartptr.i b/Examples/test-suite/director_smartptr.i
index d016af17e..166155f3a 100644
--- a/Examples/test-suite/director_smartptr.i
+++ b/Examples/test-suite/director_smartptr.i
@@ -6,7 +6,6 @@ SWIG_JAVABODY_TYPEWRAPPER(public, public, public, SWIGTYPE)
 #endif
 
 %{
-#include <boost/shared_ptr.hpp>
 #include <string>
 
 class FooBar {
diff --git a/Examples/test-suite/javascript/package.json b/Examples/test-suite/javascript/package.json
new file mode 100644
index 000000000..0967ef424
--- /dev/null
+++ b/Examples/test-suite/javascript/package.json
@@ -0,0 +1 @@
+{}
diff --git a/Lib/javascript/napi/javascriptruntime.swg b/Lib/javascript/napi/javascriptruntime.swg
index 7311aca25..f445cc887 100644
--- a/Lib/javascript/napi/javascriptruntime.swg
+++ b/Lib/javascript/napi/javascriptruntime.swg
@@ -32,7 +32,6 @@
 #error NAPI_VERSION 6 is the minimum supported target (Node.js >=14, >=12.17, >=10.20)
 #endif
 #include <napi.h>
-#include <uv.h>
 
 #include <errno.h>
 #include <limits.h>
diff --git a/configure.ac b/configure.ac
index 060c020b9..ac9c1e9e3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1193,39 +1193,36 @@ else
     fi
     AC_CHECK_PROGS(NODENPM, npm)
     JSNAPIENABLED=
+    NODE_MODULES=
     if test -n "$NODENPM"; then
-      NODENPM_PREFIX=$($NODENPM config get prefix)
+      NODE_MODULES=$($NODENPM root)
+      if test ! -d "$NODE_MODULES"; then
+        NODE_MODULES=$($NODENPM root -g)
+      fi
 
-      AC_MSG_CHECKING([for Node-API (napi) header directory])
-      if test -n "$NODENPM_PREFIX"; then
-        NODENAPI_DIR=$NODENPM_PREFIX/lib/node_modules/node-addon-api
+      if test -d "$NODE_MODULES"; then
+        AC_MSG_CHECKING([for Node-API (napi) header directory])
+        NODENAPI_DIR=$NODE_MODULES/node-addon-api
         if test -f "$NODENAPI_DIR/napi.h"; then
           JSNAPIENABLED=1
           AC_SUBST(NODENAPI_DIR)
+          AC_MSG_RESULT([$NODENAPI_DIR])
+        else
+          AC_MSG_RESULT([not found])
         fi
-      fi
-      if test -n "$JSNAPIENABLED"; then
-        AC_MSG_RESULT([$NODENAPI_DIR])
-      else
-        AC_MSG_RESULT([not found])
-      fi
-
-      AC_MSG_CHECKING([for TypeScript @types/node])
-      if test -n "$NODENPM_PREFIX"; then
-        NODETYPES_DIR=$NODENPM_PREFIX/lib/node_modules/@types
+
+        AC_MSG_CHECKING([for TypeScript @types/node])
+        NODETYPES_DIR=$NODE_MODULES/@types
         if test -f "$NODETYPES_DIR/node/index.d.ts"; then
-          true
           AC_SUBST(NODETYPES_DIR)
+          AC_MSG_RESULT([$NODETYPES_DIR])
+        else
+          AC_MSG_RESULT([not found])
         fi
       fi
-      if test -n "$JSNAPIENABLED"; then
-        AC_MSG_RESULT([$NODETYPES_DIR])
-      else
-        AC_MSG_RESULT([not found])
-      fi
     fi
     AC_SUBST(JSNAPIENABLED)
-    AC_SUBST(NODENPM_PREFIX)
+    AC_SUBST(NODE_MODULES)
   fi
 
   #----------------------------------------------------------------
@@ -1409,8 +1406,8 @@ AC_SUBST(EMSDK)
 #----------------------------------------------------------------
 
 AC_MSG_CHECKING(for emnapi)
-if test -r "${NODENPM_PREFIX}/lib/node_modules/emnapi/include/emnapi.h" ; then
-  EMNAPI_DIR="${NODENPM_PREFIX}/lib/node_modules/emnapi"
+if test -r "${NODE_MODULES}/emnapi/include/emnapi.h" ; then
+  EMNAPI_DIR="${NODE_MODULES}/emnapi"
   AC_MSG_RESULT(found in $EMNAPI_DIR)
 else
   AC_MSG_RESULT(not found)
